src/main/java/org/andersen/util/StateSaver.java

```diff
@@ -0,0 +1,32 @@
+    package org.andersen.util;
+
+    import org.andersen.config.StateConfig;
+
+    import java.io.FileOutputStream;
+    import java.io.ObjectOutputStream;
+    import java.io.IOException;
+
+    public class StateSaver {
+
+        private final FileOutputStream fileOutputStream;
+        private final ObjectOutputStream objectOutputStream;
+
+        public StateSaver(FileOutputStream fileOutputStream, ObjectOutputStream objectOutputStream) {
+            this.fileOutputStream = fileOutputStream;
+            this.objectOutputStream = objectOutputStream;
+        }
+
+        public StateSaver() throws IOException {
+            this.fileOutputStream = new FileOutputStream(StateConfig.getStateFilePath())    ;
+            this.objectOutputStream = new ObjectOutputStream(fileOutputStream);
+        }
+
+        public void saveState(Object state, String filePath) {
+            try {
+                objectOutputStream.writeObject(state);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
```

Comment:
what happened here?

Reply:
I changed only saveState method to use existing instance and not create a new one. perhaps i accidentally deleted and reverted then. i dont know why its all green

Reply:
or maybe i was deleting my test file and messed up the names and deleted the util class instead of a test one

Reply:
to be honest, i dont remember, i only remember that i changed saveState method because it created a new instance every time i called it

Reply:
whats wrong with formatting? for me everything looks clean

Reply:
could you please send a screenshot in telegram

Reply:
got it, extra tabs, not sure why